### ISSUE TITLE
Error pulling image of private registry in cloud-config

### ISSUE THREAD

**Issue by arkkanoid** [coreos/coreos-kubernetes#192]
Hi, 
I've a cloudconfig file that runs an image the first time a node is up. If this image is in the docker hub, the application runs perfectly. But when I set the image in my private registry (myregistry.com) I get this error in journalctl:

    Dec 11 08:30:13 ip-10-0-1-101.eu-west-1.compute.internal dockerd[1059]: time="2015-12-11T08:30:13.583882362Z" level=error msg="HTTP Error" err="could not find image: no such id: myregistry.com/myimage:latest" statusCode=404
    Dec 11 08:30:13 ip-10-0-1-101.eu-west-1.compute.internal dockerd[1059]: time="2015-12-11T08:30:13.620892489Z" level=info msg="POST /v1.20/images/create?fromImage=myregistry.com%2Fmyimage%3Alatest"
    Dec 11 08:30:16 ip-10-0-1-101.eu-west-1.compute.internal dockerd[1059]: time="2015-12-11T08:30:16.202247787Z" level=info msg="POST /v1.20/containers/create?name=myimage"
    Dec 11 08:30:16 ip-10-0-1-101.eu-west-1.compute.internal dockerd[1059]: time="2015-12-11T08:30:16.203002405Z" level=error msg="Handler for POST /containers/create returned error: No such image: myregistry.com/myimage:latest (tag: latest)"
    Dec 11 08:30:16 ip-10-0-1-101.eu-west-1.compute.internal dockerd[1059]: time="2015-12-11T08:30:16.203047060Z" level=error msg="HTTP Error" err="No such image: myregistry.com/myimage:latest (tag: latest)" statusCode=404
    Dec 11 08:30:16 ip-10-0-1-101.eu-west-1.compute.internal dockerd[1059]: time="2015-12-11T08:30:16.213868433Z" level=info msg="POST /v1.20/images/create?fromImage=myregistry.com%2Fmyimage&tag=latest"
    Dec 11 08:30:19 ip-10-0-1-101.eu-west-1.compute.internal systemd[1]: myimage.service: Unit entered failed state.
    Dec 11 08:30:24 ip-10-0-1-101.eu-west-1.compute.internal dockerd[1059]: time="2015-12-11T08:30:24.606905440Z" level=error msg="Handler for GET /images/{name:.*}/history returned error: could not find image: no such id: myregistry.com/myimage:latest"

When the server is up, if I execute the docker command specified in cloud-config, it works:
    docker run -d --name myimage myregistry.com/myimage:latest

Is Coreos getting the config of my private registry? 

My cloud-config file:

    "#cloud-config
    coreos:
      update:
        reboot-strategy: \"off\"
    
      flannel:
        interface: $private_ipv4
        etcd_endpoints: http://10.0.0.50:2379
    
      units:
      - name: install-worker.service
        command: start
        content: |
          [Service]
          ExecStart=/bin/bash /tmp/install-worker.sh
          Type=oneshot
    
      - name: myimage.service
        command: start
        content: |
          [Unit]
          Description=My Image
          Requires=docker.socket
          After=docker.socket
    
          [Service]
          ExecStartPre=/bin/sh -c 'docker inspect myimage >/dev/null 2>&1 && docker rm -f myimage || true
          ExecStart=/usr/bin/docker run -d --name myimage myregistry.com/myimage:latest
          Restart=on-failure
          RestartSec=5
    
    
    write_files:
      content: |
        [Service]
        Environment='DOCKER_OPTS=--insecure-registry=\"myregistry.com\"'
    
    - path: /home/core/.docker/config.json
      owner: core:core
      permissions: '0644'
      content: |
        {
         \"auths\": {
          \"myregistry.com\": {
           \"auth\": \"XXXXXXXXXXX\",
           \"email\": \"\"
         }
        }
    
    - path: /run/coreos-kubernetes/options.env
      content: |
        ETCD_ENDPOINTS=http://10.0.0.50:2379
        CONTROLLER_ENDPOINT=https://10.0.0.50
        ARTIFACT_URL=",

**Comment by robszumski** [comment]
The 404 is kind of strange. It’s hard to tell with the fake name what is going on, but are you possibly pulling from docker hub with “myregistrycom/myimage” instead of myregistry.com/myimage?

**Comment by arkkanoid** [comment]
@robszumski , I checked the domain name and it's correct. 
Copying the command I have in ExecStart and executing it manually on coreos works perfectly.

Maybe the commands listed in the cloud-config file are executed with another user than 'coreos'? Because the specific config I set for my private registry are in the 'coreos' user directory.

Another possible error I see, in: DOCKER_OPTS=--insecure-registry=\"myregistry.com\".
I don't set the 'https://', it's necessary? It would be odd because as I said it works executing the command manually.
